prefect_gcp/credentials.py

```diff
@@ -84,62 +85,107 @@ class GcpCredentials(Block):
-        service_account_file or service_account_info.
+        Ensure that only a service account file or service account info ias provided.
         """
-        if service_account_info and service_account_file:
+        if (
+            values.get("service_account_info") is not None
+            and values.get("service_account_file") is not None
+        ):
             raise ValueError(
                 "Only one of service_account_info or service_account_file "
                 "can be specified at once"
             )
-        elif service_account_file:
-            service_account_file = Path(service_account_file).expanduser()
-            if not service_account_file.exists():
-                raise ValueError("The provided path to the service account is invalid")
-            credentials = Credentials.from_service_account_file(service_account_file)
-        elif service_account_info:
-            credentials = Credentials.from_service_account_info(service_account_info)
+        return values
+
+    @validator("service_account_file")
+    def _check_service_account_file(cls, file):
+        """Get full path of provided file and make sure that it exists."""
+        if not file:
+            return file
+
+        service_account_file = Path(file).expanduser()
+        if not service_account_file.exists():
+            raise ValueError("The provided path to the service account is invalid")
+        return service_account_file
+
+    @property
+    def project_id(self):
+        if self._project_id is None:
+            if self.project:
+                self._project_id = self.project
+            else:
+                self._project_id = (
+                    self.get_credentials_from_service_account().project_id
+                )
+        return self._project_id
+
+    def get_credentials_from_service_account(self) -> Union[Credentials, None]:
+        """
+        Helper method to serialize credentials by using either
+        service_account_file or service_account_info.
+        """
+        if self.service_account_file:
+            credentials = Credentials.from_service_account_file(
+                self.service_account_file,
+                scopes=["https://www.googleapis.com/auth/cloud-platform"],
+            )
+        elif self.service_account_info:
+            credentials = Credentials.from_service_account_info(
+                self.service_account_info,
+                scopes=["https://www.googleapis.com/auth/cloud-platform"],
+            )
         else:
             return None
         return credentials
 
+    def get_access_token(self):
+        """
+        See: https://stackoverflow.com/a/69107745
+        Also: https://www.jhanley.com/google-cloud-creating-oauth-access-tokens-for-rest-api-calls/
+        """  # noqa
+        request = google.auth.transport.requests.Request()
+        credentials = self.get_credentials_from_service_account()
+        credentials.refresh(request)
```

Comment:
I would guess that this makes a network call, so it should probably be run in a worker thread and this method should be async.